js/models/conversations.js

```diff
@@ -239,37 +238,39 @@
       }
     },
     // This goes through all our message history and finds a friend request
-    // But this is not a concurrent operation and thus `updatePendingFriendRequests` is used
+    // But this is not a concurrent operation and thus updatePendingFriendRequests is used
     async hasPendingFriendRequests() {
       // Go through the messages and check for any pending friend requests
       const messages = await window.Signal.Data.getMessagesByConversation(
         this.id,
-        { 
+        {
           type: 'friend-request',
           MessageCollection: Whisper.MessageCollection,
         }
       );
-
-      for (const message of messages.models) {
-        if (message.isFriendRequest() && message.attributes.friendStatus === 'pending') return true;
-      }
-
-      return false;
+      const pendingFriendRequest =
+        messages.models.find(message =>
+          message.isFriendRequest() &&
+          message.attributes.friendStatus === 'pending'
+        );
+      return pendingFriendRequest !== undefined;
```

Comment:
Would a `!!pendingFriendRequest` suffice here?

Reply:
I think I like leaving it as is actually, easier to read imo

Reply:
I have no strong opinion, as I'm not overly familiar with the `!!` notation.
So I'd say I leave it to the committer's preference.